org.eclipse.jdt.ls.debug/src/org/eclipse/jdt/ls/debug/adapter/DebugAdapter.java

```diff
@@ -964,6 +988,188 @@ public void recyclableThreads(ThreadReference thread) {
+                    thread = ((ThreadObjectReference) obj).getThread();
+                    if (currentObj instanceof ArrayReference) {
+                        ArrayReference array = (ArrayReference) currentObj;
+                        Type eleType = ((ArrayType) array.referenceType()).componentType();
+                        newValue = setArrayValue(array, eleType, Integer.parseInt(arguments.name), arguments.value, options);
+                    } else {
+                        if (StringUtils.isBlank(belongToClass)) {
+                            Field field = currentObj.referenceType().fieldByName(name);
+                            if (field != null) {
+                                if (field.isStatic()) {
+                                    newValue = this.setStaticFieldValue(currentObj.referenceType(), field,
+                                            arguments.name, arguments.value, options);
+                                } else {
+                                    newValue = this.setObjectFieldValue(currentObj, field, arguments.name,
+                                            arguments.value, options);
+                                }
+                            } else {
+                                throw new IllegalArgumentException(
+                                        String.format("SetVariableRequest: Variable %s cannot be found.", arguments.name));
+                            }
+                        } else {
+                            newValue = setFieldValueWithConflict(currentObj, currentObj.referenceType().allFields(),
+                                    name, belongToClass, arguments.value, options);
+                        }
+                    }
+                } else {
+                    throw new IllegalArgumentException(
+                            String.format("SetVariableRequest: Variable %s cannot be found.", arguments.name));
+                }
+            } catch (IllegalArgumentException | AbsentInformationException | InvalidTypeException
+                    | UnsupportedOperationException | ClassNotLoadedException e) {
+                return new Responses.ErrorResponseBody(convertDebuggerMessageToClient(e.getMessage()));
+            }
+            int referenceId = getReferenceId(thread, newValue, showStaticVariables);
+
+            int indexedVariables = 0;
+            if (newValue instanceof ArrayReference) {
+                indexedVariables = ((ArrayReference) newValue).length();
+            }
+            return new Responses.SetVariablesResponseBody(
+                    this.variableFormatter.typeToString(newValue == null ? null : newValue.type(), options), // type
+                    this.variableFormatter.valueToString(newValue, options), // value,
+                    referenceId, indexedVariables);
+
+        }
+
+        private Value setValueProxy(Type type, String value, SetValueFunction setValueFunc, Map<String, Object> options)
```

Comment:
java.util.function.Consumer<Value> will force you to eat non-runtime exceptions at the lamp, SetValueFunction has the signature of throws jdi exceptions.